### PR TITLE
[ci] run core flaky test functions

### PR DESCRIPTION
**Pull request by can-anyscale**
As a follow up of https://github.com/ray-project/ray/pull/41195/files, we remove core flaky test targets, and run flaky test functions in the flaky job

I switch off the postmerge only conditions for testing. Will undo that change before merging.

Test:
- CI

### CODE REVIEW

File: .buildkite/core.rayci.yml [hunk 1]
```diff
@@ -241,10 +241,18 @@ steps:
     tags: python
     instance_type: large
     # run only on postmerge
-    if: build.env("BUILDKITE_PIPELINE_ID") == "0189e759-8c96-4302-b6b5-b4274406bf89"
+    # if: build.env("BUILDKITE_PIPELINE_ID") == "0189e759-8c96-4302-b6b5-b4274406bf89"
```
Comment (can-anyscale) [inline .buildkite/core.rayci.yml:244]: will undo this change before merging

File: ci/ray_ci/core.tests.yml [hunk 2]
```diff
@@ -1,8 +1,3 @@
 flaky_tests:
-  - //python/ray/tests:test_placement_group
-  - //python/ray/tests:test_runtime_env_working_dir_3
-  - //python/ray/tests:test_state_api_log
-  - //python/ray/tests:test_client_builder
   - //python/ray/tests:test_client_library_integration
```
Comment (rickyyx) [inline ci/ray_ci/core.tests.yml:2]: ah- i actually wasn't able to find any flaky job run with this one. @rkooo567 are you aware of any?

**Review by rickyyx — APPROVED** [review]
